submarine-commons/cluster-server/src/main/java/org.apache.submarine/utils/SubmarineConfiguration.java

```diff
@@ -0,0 +1,559 @@
+    return getBoolean(ConfVars.SERVER_SSL);
+  }
+
+  public int getServerPort() {
+    return getInt(ConfVars.SERVER_PORT);
+  }
+
+  public int getServerSslPort() {
+    return getInt(ConfVars.SERVER_SSL_PORT);
+  }
+
+  public String getKeyStorePath() {
+    String path = getString(ConfVars.SSL_KEYSTORE_PATH);
+    return path;
+  }
+
+  public String getKeyStoreType() {
+    return getString(ConfVars.SERVER_SSL_KEYSTORE_TYPE);
+  }
+
+  public String getKeyStorePassword() {
+    return getString(ConfVars.SERVER_SSL_KEYSTORE_PASSWORD);
+  }
+
+  public String getKeyManagerPassword() {
+    String password = getString(ConfVars.SERVER_SSL_KEY_MANAGER_PASSWORD);
+    if (password == null) {
+      return getKeyStorePassword();
+    } else {
+      return password;
+    }
+  }
+
+  public boolean useClientAuth() {
+    return getBoolean(ConfVars.SSL_CLIENT_AUTH);
+  }
+
+  public String getTrustStorePath() {
+    String path = getString(ConfVars.SERVER_SSL_TRUSTSTORE_PATH);
+    if (path == null) {
+      path = getKeyStorePath();
+    }
+    return path;
+  }
+
+  public String getTrustStoreType() {
+    String type = getString(ConfVars.SERVER_SSL_TRUSTSTORE_TYPE);
+    if (type == null) {
+      return getKeyStoreType();
+    } else {
+      return type;
+    }
+  }
+
+  public String getTrustStorePassword() {
+    String password = getString(ConfVars.SERVER_SSL_TRUSTSTORE_PASSWORD);
+    if (password == null) {
+      return getKeyStorePassword();
+    } else {
+      return password;
+    }
+  }
+
+  public Integer getJettyRequestHeaderSize() {
+    return getInt(ConfVars.SERVER_JETTY_REQUEST_HEADER_SIZE);
+  }
+
+  public String getRelativeDir(ConfVars c) {
+    return getRelativeDir(getString(c));
+  }
+
+  public String getRelativeDir(String path) {
+    if (path != null && path.startsWith(File.separator) || isWindowsPath(path)) {
+      return path;
+    } else {
+      return getString("./") + File.separator + path;
+    }
+  }
+
+  public boolean isWindowsPath(String path) {
+    return path.matches("^[A-Za-z]:\\\\.*");
+  }
+
+  public String getJdbcDriverClassName() {
+    return getString(ConfVars.JDBC_DRIVERCLASSNAME);
+  }
+
+  public String getJdbcUrl() {
+    return getString(ConfVars.JDBC_URL);
+  }
+
+  @VisibleForTesting
+  public void setJdbcUrl(String testJdbcUrl) {
+    properties.put(ConfVars.JDBC_URL.getVarName(), testJdbcUrl);
+  }
+
+  public String getJdbcUserName() {
+    return getString(ConfVars.JDBC_USERNAME);
+  }
+
+  @VisibleForTesting
+  public void setJdbcUserName(String userName) {
+    properties.put(ConfVars.JDBC_USERNAME.getVarName(), userName);
+  }
+
+  public String getJdbcPassword() {
+    return getString(ConfVars.JDBC_PASSWORD);
+  }
+
+  @VisibleForTesting
+  public void setJdbcPassword(String password) {
+    properties.put(ConfVars.JDBC_PASSWORD.getVarName(), password);
+  }
+
+  public String getClusterAddress() {
+    return getString(ConfVars.WORKBENCH_CLUSTER_ADDR);
```

Comment:
I think we'd better move the workbench related code to `submarine-workbench`. Such as create the `WorkbechConfiguration` and convert the `SubmarineConfiguration` to abstract.

Reply:
Submarine has only one configuration file, only one configuration management class. Now the `SubmarineConfiguration` class has been moved to the `commons-utils` module to become a shared module.